docs/topics/classes.md

```diff
@@ -1,245 +1,747 @@
+inheriting its properties and functions while adding or modifying behavior.
+
+> For detailed information about inheritance hierarchies and the use of the `open` keyword, see the [Inheritance](inheritance.md) section.
+> 
+{style="note"}
 
 ## Abstract classes
 
-A class may be declared `abstract`, along with some or all of its members.
-An abstract member does not have an implementation in its class.
-You don't need to annotate abstract classes or functions with `open`.
+An abstract class in Kotlin is a class that cannot be instantiated directly and is designed to be inherited 
+by other classes. 
+
+An abstract class can define abstract properties and functions, which must be implemented 
+by subclasses.
+
+You can declare an abstract class using the `abstract` modifier:
+
+```kotlin
+abstract class Person
+```
+
+Like any other class, abstract classes can also have constructors. 
+These constructors initialize class properties and enforce required parameters for subclasses:
+
+```kotlin
+abstract class Person(val name: String, val age: Int)
+```
+
+An abstract class can have both abstract and non-abstract members (properties and functions). 
+To declare a member as abstract, you must use the `abstract` keyword explicitly.
+
+Abstract members do not have an implementation in the abstract class. The `override`
+keyword ensures that a subclass provides an actual implementation for an abstract function:
 
 ```kotlin
-abstract class Polygon {
-    abstract fun draw()
+// Abstract class with a primary constructor that defines 'name' and 'age'
+abstract class Person(
+    val name: String,
+    val age: Int
+) {
+    // Abstract member (doesn't provide default implementation, must be implemented by subclasses)
+    abstract fun introduce()
+
+    // Non-abstract member (has a default implementation)
+    fun greet() {
+        println("Hello, my name is $name.")
+        // Hello, my name is Alice.
+    }
 }
 
-class Rectangle : Polygon() {
-    override fun draw() {
-        // draw the rectangle
+// Subclass that provides an implementation for the abstract member
+class Student(
+    name: String,
+    age: Int,
+    val school: String
+) : Person(name, age) {
+    override fun introduce() {
+        println("I am $name, $age years old, and I study at $school.")
+        // I am Alice, 20 years old, and I study at Engineering University.
     }
 }
+
+fun main() {
+    // Creates an object of the Student class
+    val student = Student("Alice", 20, "Engineering University")
+    // Calls the non-abstract member
+    student.greet()
+    // Calls the overridden abstract member
+    student.introduce()
+}
 ```
+{kotlin-runnable="true"}
+
+You don't need to annotate abstract classes or functions with the `open` keyword because they are implicitly 
+open and designed to be inherited. 
+
+Similarly, you can override a non-abstract `open` member with an abstract one:
 
-You can override a non-abstract `open` member with an abstract one.
+## Open keyword
```

Comment:
I would say this section makes more sense in the `Inheritance` part of the documentation